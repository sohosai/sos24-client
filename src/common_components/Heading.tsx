import { FC, PropsWithChildren } from "react";
import { flex } from "@styled-system/patterns";
import Image from "next/image";
import triangleIcon from "@/assets/Triangle.svg?url";
import { css } from "@styled-system/css";

export const Heading: FC<PropsWithChildren> = ({ children }) => {
  return (
    <div>
      <h3
        className={flex({
          marginTop: 8,
          gap: 3,
        })}>
        <Image src={triangleIcon} alt="" />
        <span
          className={css({
            fontSize: "md",
            fontWeight: "bold",
          })}>
          {children}
        </span>
      </h3>
    </div>
  );
};
