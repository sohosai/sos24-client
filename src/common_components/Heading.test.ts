import { expect, test } from "vitest";
import { composeStories } from "@storybook/react";

import * as stories from "./Heading.stories";

const { Default } = composeStories(stories);

test("renders and executes the play function", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
