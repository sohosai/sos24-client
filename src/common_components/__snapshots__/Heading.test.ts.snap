// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders and executes the play function 1`] = `
<div>
  <div>
    <h3
      class="d_flex mt_8 gap_3"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="1"
        loading="lazy"
        src="src/assets/Triangle.svg"
        style="color: transparent;"
        width="1"
      />
      <span
        class="fs_md font_bold"
      >
        Heading
      </span>
    </h3>
  </div>
</div>
`;
